apps/application/flow/step_node/ai_chat_step_node/impl/base_chat_node.py

```diff
@@ -73,6 +73,10 @@ def get_default_model_params_setting(model_id):
 
 
 class BaseChatNode(IChatNode):
+    def save_context(self, details, workflow_manage):
+        self.context['answer'] = details.get('answer')
+        self.context['question'] = details.get('question')
+
     def execute(self, model_id, system, prompt, dialogue_number, history_chat_record, stream, chat_id, chat_record_id,
                 model_params_setting=None,
                 **kwargs) -> NodeResult:
```

Comment:
Code has been cleaned up and any inconsistencies or errors are resolved but the logic is somewhat complicated. A key issue here is that if you want to handle multiple conversations, it might be better not just storing 'answer' in `self.context` which can lead to potential issues when dealing with asynchronous requests (like from different servers). It would likely make more sense for each conversation context to contain its own set of values.

In addition, since you're using both `'history_chat_record'` and `stream` arguments in your method signature without a description/clarification within, these parameters can either be used interchangeably or possibly explained clearly for clarity.